app/src/main/java/io/github/gsantner/memetastic/ui/MemeItemAdapter.java

```diff
@@ -166,6 +227,26 @@ private void toggleFavorite(ViewHolder holder) {
         }
     }
 
+    private void toggleHidden(ViewHolder holder, int position){
+        MemeData.Image image = (MemeData.Image) holder.imageView.getTag();
+        String filePath = image.fullPath.getAbsolutePath();
+
+        if (_app.settings.toggleHiddenMeme(filePath)){
+            image.isTemplate = false;
+            _imageDataList.remove(image);
+            notifyItemRemoved(position);
+        }else {
+            image.isTemplate = true;
```

Comment:
if the image is a template should not be controlled in any way by the state of hidde

Reply:
Okay, will removing toggle the field

Reply:
Just to be sure: I meant that you should not control `isTemplate` by Hidden